src/compounds/tabs/src/index.tsx

```diff
@@ -63,34 +62,164 @@ interface TabsProps extends React.HTMLAttributes<HTMLDivElement> {
 
 export const Tabs: React.FC<TabsProps> = ({ children }) => {
   const [activeTab, setActiveTab] = useState(0)
+  const [left, setLeft] = useState(0)
+  const tabs: RefObject<HTMLDivElement> = React.createRef()
+  const tabWrap: RefObject<HTMLDivElement> = React.createRef()
+  const [padding, setPadding] = useState(0)
+  const [width, setWidth] = useState(0)
+  const [wrapWidth, setWrapWidth] = useState(0)
+  const [isPressed, setIsPressed] = useState(false)
+  const [lastX, setLastX] = useState(0)
+  const [showLeftBorder, setShowLeftBorder] = useState(false)
+  const [showRightBorder, setShowRightBorder] = useState(false)
+  const [scrollEnd, setScrollEnd] = useState(false)
+
+  useEffect(() => {
+    setPadding(tabs?.current?.offsetHeight)
+    setWidth(tabs?.current?.scrollWidth)
+  }, [tabs])
+
+  useEffect(() => {
+    setWrapWidth(tabWrap?.current?.offsetWidth)
+
+    if (width > wrapWidth && !scrollEnd) {
+      setShowRightBorder(true)
+    }
+
+    if (left !== 0) {
+      setShowLeftBorder(true)
+    } else {
+      setShowLeftBorder(false)
+    }
+  }, [tabWrap])
+
+  const setRealLeft = (left: number) => {
+    if (left > 0) {
+      return setLeft(0)
+    }
+
+    const minLeft = -(width - wrapWidth)
+
+    if (left <= minLeft) {
+      setScrollEnd(true)
+      setShowRightBorder(false)
+      return setLeft(minLeft)
+    } else {
+      setScrollEnd(false)
+      setShowRightBorder(true)
+    }
+
+    setLeft(left)
+  }
+
+  const onMouseDown = (event: any) => {
+    if (wrapWidth > width) return
+
+    setLastX(event.clientX)
+    setIsPressed(true)
+  }
+
+  const onMouseUp = () => {
+    setIsPressed(false)
+  }
+
+  const onMouseMove = (event: any) => {
+    if (!isPressed) return
+
+    if (lastX > event.clientX) {
+      // drag left
+      const diff = lastX - event.clientX
+      setRealLeft(left - diff)
+    } else {
+      // drag right
+      const diff = event.clientX - lastX
+
+      setRealLeft(left + diff)
+    }
+
+    setLastX(event.clientX)
+  }
+
+  const onTouchStart = (event: any) => {
+    if (wrapWidth > width) return
+    setLastX(event.touches[0].clientX)
+    setIsPressed(true)
+  }
+
+  const onTouchEnd = () => {
+    setIsPressed(false)
+  }
+
+  const onTouchMove = (event: any) => {
+    if (!isPressed) return
+
+    if (lastX > event.touches[0].clientX) {
+      const diff = lastX - event.touches[0].clientX
+      setRealLeft(left - diff)
+    } else {
+      const diff = event.touches[0].clientX - lastX
+      setRealLeft(left + diff)
+    }
+
+    setLastX(event.touches[0].clientX)
+  }
+
+  const onDragStart = (event: any) => {
+    event.preventDefault()
+  }
+
+  window.addEventListener('mouseup', onMouseUp)
```

Comment:
is something similar needed for on touch up?

Reply:
no dont think so as this was to fix an IE issue and no IE devices have touch to my knowledge

Reply:
i was there when we did this fix, it was for when the mouse moves outside the container before mouse up